template/src/ontology/run.sh.jinja2

```diff
@@ -63,7 +63,14 @@ if [ -n "$TAG_IN_IMAGE" ]; then
   ODK_IMAGE=$(echo $ODK_IMAGE | awk -F':' '{ print $1 }')
 fi
 ODK_TAG=${ODK_TAG:-latest}
-ODK_JAVA_OPTS={% if project.robot_java_args is defined and project.robot_java_args|length %}${ODK_JAVA_OPTS:-{{ project.robot_java_args }}}{% else %}${ODK_JAVA_OPTS:--Xmx8G}{% endif %}
+{% if project.robot_java_args is defined and project.robot_java_args|length %}
+ODK_JAVA_OPTS=${ODK_JAVA_OPTS:-{{ project.robot_java_args }}}
+{% else %}
+if [ -z "$USE_SINGULARITY" ]; then
+    {% raw %}DEFAULT_MAX_MEM=$(bc <<<"($(docker info --format={{.MemTotal}}) * .9) / (1024*1024*1024)")G{% endraw %}
```

Comment:
I have not reviewed this, assuming this was tested.

Reply:
Tested on both my GNU/Linux and macOS machines.

(And before you ask: `bc` is not a GNUism, it’s a POSIX-specified command. :) )

Of note, this returns a value in gigabytes, _rounded to the lower gigabyte_. This means that if the maximum amount of memory allocated to Docker is barely above 1G (so that 90% of it is less than 1G), this will return `0G`.

I considered adding an explicit test for that condition, but decided it was not worth it. I don’t expect anyone to run the ODK with ~1G of memory anyway (not even sure it _can_ run with 1G) -- and if someone _really_ wants to try that, they can always explicitly set ODK_JAVA_OPTS.

Reply:
Super, thanks for the explanation!